client/src/components/Body/index.js

```diff
@@ -0,0 +1,49 @@
+import { Grid, Paper, makeStyles } from "@material-ui/core";
+import SelectedWidgetsModal from "../SelectWidgetsModal";
+import styles from "./style.module.scss";
+import createPersistedState from "use-persisted-state";
+import { useEffect } from "react";
+
+const useWidgetState = createPersistedState("selectedWidgets");
```

Comment:
Could you put this in App.js and pass 'selectedWidgets' and 'setSelectedWidgets' in as props?